AlphaWallet/Core/Types/Tokens/TokenRepairService.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  TokenRepairService.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 08.08.2022.
+//
+
+import Foundation
+import Combine
+
+fileprivate extension Token {
+    var isSpuriousToken: Bool {
+        switch type {
+        case .nativeCryptocurrency, .erc20:
+            return name.isEmpty && symbol.isEmpty && decimals == 0
+        case .erc1155, .erc875, .erc721, .erc721ForTickets:
+            return name.isEmpty || symbol.isEmpty
+        }
+    }
+}
+final class TokenRepairService {
+    enum RepairTokenError: Error {
+        case inner(PromiseError)
+        case wrongResponse
+    }
+
+    private let tokensDataStore: TokensDataStore
+    private let sessionsProvider: SessionsProvider
+    private let importToken: ImportToken
+    private var cancelable = Set<AnyCancellable>()
+    private var operations: [AddressAndRPCServer: AnyCancellable] = [:]
+    private let queue = DispatchQueue(label: "org.alphawallet.swift.\(String(describing: TokenRepairService.self))", qos: .utility)
+
+    init(tokensDataStore: TokensDataStore, sessionsProvider: SessionsProvider, importToken: ImportToken) {
+        self.tokensDataStore = tokensDataStore
+        self.sessionsProvider = sessionsProvider
+        self.importToken = importToken
+    }
+
+    func start() {
+        let spuriousTokens = tokensDataStore.enabledTokensChangeset(for: Array(sessionsProvider.activeSessions.keys))
```

Comment:
I don't understand how this works?

How does it detect/filter the tokens that need repairing? What is the criteria?